### ISSUE TITLE
[Bug] When heterograph have ``dgl.NTYPE`` field, the node type of to_homo() will mismatch the ``dgl.NTYPE`` field.

### ISSUE THREAD

**Issue by classicsong** [dmlc/dgl#1719]
## To Reproduce

```
>>> import dgl
>>> import torch as th
>>> from dgl.data.rdf import AIFB

>>> dataset = AIFB()
>>> hg = dataset.graph
>>> num_ntype = len(hg.ntypes)
>>> ntid_cnt = {}
>>> for i, ntype in enumerate(hg.ntypes):
>>>     print('NTID {} ({}) has nodes {}'.format(hg.nodes[ntype].data[dgl.NTYPE][0],
>>>                                              ntype,
>>>                                              hg.number_of_nodes(ntype)))
NTID 0 (Forschungsgebiete) has nodes 146
NTID 4 (Forschungsgruppen) has nodes 5
NTID 6 (Kooperationen) has nodes 28
NTID 2 (Personen) has nodes 237
NTID 1 (Projekte) has nodes 78
NTID 5 (Publikationen) has nodes 1318
NTID 3 (_Literal) has nodes 5450

>>> g = dgl.to_homo(hg)
>>> node_tids = g.ndata[dgl.NTYPE]
>>> for i in range(num_ntype):
>>>    print('NTID {} has nodes {}'.format(i,
>>>                                        th.nonzero(node_tids==i).size()))
NTID 0 has nodes torch.Size([146, 1])
NTID 1 has nodes torch.Size([5, 1])
NTID 2 has nodes torch.Size([28, 1])
NTID 3 has nodes torch.Size([237, 1])
NTID 4 has nodes torch.Size([78, 1])
NTID 5 has nodes torch.Size([1318, 1])
NTID 6 has nodes torch.Size([5450, 1])
```
The number of nodes for some of the NTIDs mismatch before/after to_homo

## Expected behavior

The number of nodes for any NTIDs must match before/after to_homo

## Environment

 - DGL Version (e.g., 1.0):
 - Backend Library & Version (e.g., PyTorch 0.4.1, MXNet/Gluon 1.3):
 - OS (e.g., Linux):
 - How you installed DGL (`conda`, `pip`, source):
 - Build command you used (if compiling from source):
 - Python version:
 - CUDA/cuDNN version (if applicable):
 - GPU models and configuration (e.g. V100):
 - Any other relevant information:

**Comment by BarclayII** [comment]
Since we have `NTYPE` field in the graph, I assume it is converted from a homogeneous graph.  Why did `NTYPE` field not match the node types in the original dataset?

**Comment by classicsong** [comment]
> Since we have `NTYPE` field in the graph, I assume it is converted from a homogeneous graph. Why did `NTYPE` field not match the node types in the original dataset?

The to_homo() ignore the `NTYPE` field when converting heterograph to homograph.

**Comment by BarclayII** [comment]
Why should `to_homo` care about the `NTYPE` field?  The code above looks correct to me.

**Comment by BarclayII** [comment]
Also I feel that `dgl.NTYPE` should be excluded from feature copy in `dgl.to_homo()`

**Comment by classicsong** [comment]
This is fixed in 0.5.0.
